codalab/rest/bundles.py

```diff
@@ -377,6 +378,21 @@ def _set_bundle_permissions():
     return BundlePermissionSchema(many=True).dump(new_permissions).data
 
 
+@get('/bundles/locations', apply=AuthenticatedProtectedPlugin())
+def _fetch_locations():
+    """
+    Fetch locations of bundles.
+
+    Query parameters:
+    - `uuids`: List of bundle UUID's to get the locations for
+    """
+    bundle_uuids = query_get_list('uuids')
+    uuids_to_locations = {
+        uuid: local.bundle_store.get_bundle_location(uuid) for uuid in bundle_uuids
```

Comment:
Once the link PR gets in, this will need to be changed to something like 

```python
bundle_link_urls = local.model.get_bundle_metadata(bundle_uuids, "link_url")
uuids_to_locations = {
        uuid: bundle_link_urls.get("uuid") or local.bundle_store.get_bundle_location(uuid) for uuid in bundle_uuids
}
```